core/src/main/java/org/apache/iceberg/util/SnapshotUtil.java

```diff
@@ -79,8 +79,11 @@ public static Snapshot oldestAncestorAfter(Table table, Long timestampMillis) {
       return null;
     }
 
-    while (current.parentId() != null && table.snapshot(current.parentId()).timestampMillis() >= timestamp) {
-      current = table.snapshot(current.parentId());
+    for (Snapshot snapshot : table.snapshots()) {
```

Comment:
I think you still need to do the "parent" look up here since I believe this will pick up uncommitted and snapshots which have been rolled back. Ie snapshots which exist but are not part of the current history

Reply:
The issue of "parent" look up is that once some of the snapshot expired in the "parent" look up chain, the chain will be break. I am trying to figure out how we can find next available parent snapshot. For example we have a parent snapshot shot chain:
A  <- B(Expired) <- C.
C.Parent is not available in the snapshots anymore, so I am trying to figure out how we can get A as the next available parent snapshot